x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/rules/logical/PushDownAndCombineFiltersTests.java

```diff
@@ -830,14 +831,142 @@ public void testPushDownFilterPastTwoLookupJoins() {
         assertThat(rightFilter.condition().toString(), is("language_name > \"a\""));
     }
 
-    private Join createLeftJoin() {
+    private Join createLeftJoinOnFields() {
         FieldAttribute a = getFieldAttribute("a");
         FieldAttribute b = getFieldAttribute("b");
         FieldAttribute c = getFieldAttribute("c");
         EsRelation left = relation(List.of(a, b));
         EsRelation right = relation(List.of(c, b));
 
-        JoinConfig joinConfig = new JoinConfig(JoinTypes.LEFT, List.of(b), List.of(a, b), List.of(b, c));
+        JoinConfig joinConfig = new JoinConfig(JoinTypes.LEFT, List.of(a, b), List.of(b, c), null);
         return new Join(EMPTY, left, right, joinConfig);
     }
+
+    private Join createLeftJoinOnExpression() {
+        FieldAttribute a = getFieldAttribute("a");
+        FieldAttribute b1 = getFieldAttribute("b1");
+        FieldAttribute b2 = getFieldAttribute("b2");
+        FieldAttribute c = getFieldAttribute("c");
+        EsRelation left = relation(List.of(a, b1));
+        EsRelation right = relation(List.of(c, b2));
+        Expression joinOnCondition = new GreaterThanOrEqual(Source.EMPTY, b1, b2);
```

Comment:
The condition doesn't include all left and right fields. That's not problematic for this test, but may confuse others as we're unit-testing with somewhat invalid state.